doc/adr/0010-atomic-composite-grapth-operations.md

```diff
@@ -0,0 +1,267 @@
+
+Consequence Event is 2 things:
+- Trigger Event (like `:paste`)
+- Consequence Events List
+    - Containing however many events to include into transaction resolution.
+    - These events can be more of Consequence Events, or ((⚛️ Atomic Graph Ops)).
+
+
+This will allow for easier Temporality,
+We'll be able to represent Main Action that moved Knowledge Graph forward.
+
+It will maintain information about `:paste` events that injected sub-tree with ((⚛️ Atomic Graph Ops)).
+
+## Consequences
+
+We'll have additional work of Porting Semantic Events to use Atomic Graph Ops and compositions.
+
+We have working `:paste` & `:block/save` events, they are broken not ported now.
+
+We'll have smaller amount of tests in order to provide correctness guarantees
+
+
+## Additional Resources
+
+### Catalog of operations
+
+- Types of Graph Operations
+    - ⚛️ Atomic Graph Ops
+        - Not divisible Graph Ops
+        - Operations like create new block, create page, save block
+    - ⎄ Composite Graph Ops
+        - Collection of events to be executed on the graph
+        - Like `:block/save` when new link is discovered, should produce also `:page/create` event
+
+- List of Atomic Graph Operations:
+    - Page Ops
+        - ⚛️**`:page/new`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `title` - Page title page to be created
+                - `page-uid` - `:block/uid` of page to be created
+                - `block-uid` - `:block/uid` of 1st block to be created in page to be created
+        - ⚛️**`:page/rename`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `page-uid` - `:block/uid` of page to be renamed
+                - `new-name` - Page should have this name after operation
+                - `old-name` - ^^To Remove^^ This is accidental, and shouldn't be provided
+        - ⚛️**`:page/merge`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `page-uid` - `:block/uid` of page to be merged into `new-page`
+                - `new-page` - page name of a page we'll merge contents of `page-uid` page into
+                - `old-name` - ^^To Remove^^ This is accidental, and shouldn't be provided
+        - ⚛️**`:page/remove`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `page-uid` - `:block/uid` of the page to be deleted
+    - Block Ops
+        - ⚛️**`:block/new`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `parent-uid` - `:block/uid` of parent block (or page)
+                - `block-uid` - `:block/uid` of new block to be created
+                - `block-order` - `:block/order` of new block to be created
+                    - Currently it's only `int`
+                    - We could extend it to allow `int` and 2 keywords `:first` & `:last` (to say that we want this new block to be 1st among the children of `parent-uid` or last)
```

Comment:
`:after` and `:before` would also make it much more semantically meaningful in concurrent scenarios.

Reply:
Yes, thought this would be next to order, so additional attribute.
Maybe 🤔